src/Internal/Cardano/TextEnvelope.purs

```diff
@@ -100,3 +114,25 @@ textEnvelopeBytes json ty = do
   unless (te.type_ == ty) $ throwError $ JsonDecodeError $ TypeMismatch $
     show ty
   pure te.bytes
+
+_plutusScriptVxFromEnvelope
+  :: TextEnvelopeType
+  -> (ByteArray -> PlutusScript)
+  -> String
+  -> Either TextEnvelopeDecodeError PlutusScript
+_plutusScriptVxFromEnvelope version bytesToScript str = do
+  bytes <- textEnvelopeBytes str version
+  bimap
+    (CSLError <<< match { fromBytesError: \x -> x })
+    (bytesToScript <<< ST.bytes)
+    (fromBytes' bytes :: E (FromBytesError + ()) ST.PlutusScript)
+
+plutusScriptV1FromEnvelope
+  :: String -> Either TextEnvelopeDecodeError PlutusScript
```

Comment:
This interface is overcomplicated. I don't think the average user cares about the reason envelope parsing has failed and wants to pattern match on the reason. Please change it to just `Maybe` here and everywhere.

Reply:
Remove the lifting function as well, because we already have lifting for Maybe

Reply:
And the error type itself